lib/array_intersection.rb

```diff
@@ -1,6 +1,42 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(nlogn) + O(logn) where n is the number of elem is array1
+# Space complexity:O(n) --> O(n) + O(n) + O(m) where n is the number of elems in array1 and m is the number of elems in intersect
 def intersection(array1, array2)
-  raise NotImplementedError
+  intersect = []
+  return intersect if !array1 || !array2
+  array1 = merge_sort(array1)
+  array2.each do |elem|
+    intersect << elem if binary_search(array1, elem)
+  end
+  return intersect
+end
+
+def merge_sort(array)
+  return array if array.length <= 1
+  mid = array.length / 2
+  arr1 = merge_sort(array[0...mid])
```

Comment:
One note here, `array[0..mid]` is creating a new array and thus you are creating a new array with each iteration and this will impact time complexity and space complexity.